common/views/components/IIIFViewerPrototype/IIIFViewerPrototype.tsx

```diff
@@ -229,6 +241,11 @@ const IIIFViewerPrototype: FunctionComponent<IIIFViewerProps> = ({
     image => image.canvasIndex === 0
   );
   const firstRotation = firstRotatedImage ? firstRotatedImage.rotation : 0;
+  const activeIndexRef = useRef(activeIndex);
+
+  useEffect(() => {
+    activeIndexRef.current = activeIndex;
+  }, [activeIndex]);
 
   useEffect(() => {
```

Comment:
Should we pull this out into a hook and allow the component that needs it to just use the hook rather than pass it down as a prop?

Reply:
I'm putting it in the `ItemViewerContext` since it is specific to the `mainArea` of the item viewer (that's what the `ResizeObserver` is listening on). The places it's being used as props here are for the styled-components, so I think those are still required. I'm not sure making it into a hook helps much, but I could have missed something?